marshmallow/schema.py

```diff
@@ -533,10 +529,10 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
                 self.__error_handler__(errors, obj)
             exc = ValidationError(
                 errors,
-                field_names=self._marshal.error_field_names,
-                fields=self._marshal.error_fields,
+                field_names=marshal.error_field_names,
+                fields=marshal.error_fields,
```

Comment:
`error_field_names` and `error_fields` should also be available on the `ValidationError` instance.